modules.d/01systemd-bsod/module-setup.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# This file is part of dracut.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Prerequisite check(s) for module.
+check() {
+    # If the binary(s) requirements are not fulfilled the module can't be installed
+    require_binaries "$systemdutildir"/systemd-bsod || return 1
+
+    # Return 255 to only include the module, if another module requires it.
+    return 255
+}
+
+# Module dependency requirements.
+depends() {
+    # This module has external dependency on other module(s).
+    echo systemd
+    # Return 0 to include the dependent module(s) in the initramfs.
+    return 0
+}
+
+# Install the required file(s) for the module in the initramfs.
+install() {
+    inst_multiple -o \
+        "$systemdsystemunitdir"/systemd-bsod.service \
+        "$systemdsystemunitdir"/sysinit.target.wants/systemd-bsod.service \
+        "$systemdutildir"/systemd-bsod \
+        qrencode
```

Comment:
`systemd-bsod` does not require or use the `qrencode` binary, it calls `libqrencode` via dlopen, so you should add it with `inst_libdir_file`.